### PR TITLE
feat: add switching context button

### LINKED ISSUE #4920: Kube Config UI: Add button to change current context in settings
### Is your enhancement related to a problem? Please describe

Part of https://github.com/containers/podman-desktop/issues/4618

### Describe the solution you'd like

Be able to press on a context and have it change the file / also sync with the status bar.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by cdrage**
feat: add switching context button

### What does this PR do?

* Adds the ability to switch contexts / save to file by clicking on a
  button.
* Changes also propagate to the status bar

### Screenshot/screencast of this PR

https://github.com/containers/podman-desktop/assets/6422176/99303194-dceb-45af-a8d5-94312c505f85

### What issues does this PR fix or reference?

Closes https://github.com/containers/podman-desktop/issues/4920

### How to test this PR?

Go to the Kubernetes page and select another context, you should see
your file change (.kube/config) as well as the context change in the
status bar.

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

**Comment by cdrage** [comment]
Some notes:

https://github.com/containers/podman-desktop/pull/4921 is required to be merged in before this PR.


@mairin @deboer-tim so with the current penpot mockup we do not have any indication on how to "switch" between the contexts (unless it's the `<` button?):

Screenshot 2023-11-21 at 3 27 13 PM

Since we now have the "trash" icon instead, I'm unsure what to do UI wise.

TODO:

- [ ] Once UI design finalized, add rendering test.

**Comment by deboer-tim** [comment]
I like this approach (button beside delete), but the icon isn't working for me. Maybe circle-check instead, or there's something better for 'set current context'...

**Comment by lstocchi** [comment]
> I like this approach (button beside delete), but the icon isn't working for me. Maybe circle-check instead, or there's something better for 'set current context'...

I agree, at a first look i thought that the check icon was marking the active context. 
Another idea to customize https://fontawesome.com/icons/right-to-bracket?f=classic&s=solid which shows you're entering another context

**Comment by cdrage** [comment]
Screenshot 2023-11-22 at 8 54 08 AM

Updated to the "right to bracket" button.

We can do this as a temporary measure until we possibly get a "dropdown" menu as we start implementing more features for the context page?

**Comment by deboer-tim** [comment]
That's better. I don't love this icon either :) but it's fine for now.

Can we make the button disabled with button tooltip (something along the lines of 'already set as the default context') when it is the current context (vs hidden)? I think this will help reenforce the current context (and help with usability/accessibility).

I still prefer a badge for current context vs moving the title.

File: packages/renderer/src/lib/preferences/PreferencesKubernetesContextsRendering.svelte [hunk 1]
```diff
@@ -16,6 +16,17 @@ onMount(async () => {
   currentContextName = await window.kubernetesGetCurrentContextName();
 });
 
+async function handleSetContext(contextName: string) {
+  $kubernetesContexts = clearKubeUIContextErrors($kubernetesContexts);
+  try {
+    await window.kubernetesSetContext(contextName);
+  } catch (e: unknown) {
+    if (e instanceof Error) {
```
Comment (jeffmaury) [inline packages/renderer/src/lib/preferences/PreferencesKubernetesContextsRendering.svelte:24]: What if e is not an Error ? But it does not seem to be related to this PR

File: packages/renderer/src/lib/preferences/PreferencesKubernetesContextsRendering.svelte [hunk 2]
```diff
@@ -73,6 +84,13 @@ async function handleDeleteContext(contextName: string) {
                 <span class="text-md" aria-label="context-name">{context.name}</span>
               </div>
             </div>
+            <!-- Only show the set context button if it is not the current context -->
+            {#if !context.currentContext}
+              <ListItemButtonIcon
+                title="Set Context"
```
Comment (jeffmaury) [inline packages/renderer/src/lib/preferences/PreferencesKubernetesContextsRendering.svelte]: Quite vague: may be something like Set as current context ?

**Review by jeffmaury — APPROVED** [review]
LGTM

File: packages/renderer/src/lib/preferences/PreferencesKubernetesContextsRendering.svelte (hunk 1 above)
Comment (cdrage) [reply packages/renderer/src/lib/preferences/PreferencesKubernetesContextsRendering.svelte:24]: You're right, not part of the PR

**Comment by cdrage** [comment]
> That's better. I don't love this icon either :) but it's fine for now.
> 
> Can we make the button disabled with button tooltip (something along the lines of 'already set as the default context') when it is the current context (vs hidden)? I think this will help reenforce the current context (and help with usability/accessibility).
> 
> I still prefer a badge for current context vs moving the title.

I agree :) We can take this up in https://github.com/containers/podman-desktop/issues/4889 with improvements as we finalize the modified context UI.

**Review by lstocchi — COMMENTED** [review]
Should we send any event when we change the context?
Let's say i'm in the deployment page, then switch context, the deployment store should be updated. By listening to that event we can force a refresh

**Comment by lstocchi** [comment]
> Should we send any event when we change the context? Let's say i'm in the deployment page, then switch context, the deployment store should be updated. By listening to that event we can force a refresh

Maybe not related to this PR, more about the statusbar

**Review by lstocchi — APPROVED** [review]
LGTM, works fine
